src/lib/corchuelo.rs

```diff
@@ -183,22 +178,28 @@ pub(crate) fn recover<TokId: Clone + Copy + Debug + TryFrom<usize> + TryInto<usi
                     }
                 }
 
+                // As described, at this point we should add (new_la_idx - la_idx) Shifts to the
+                // repair sequence. However, there's no point in doing this if they're added to a
+                // finisher: Shifts at the end of a repair sequence confuse users and slow down
+                // parsing. We thus only add Shifts if this is a non-finisher.
+
+                let sc = score(&repairs); // Since Shifts don't count to the score, this isn't
```

Comment:
I'm not sure I understand why this line was moved outside of `if finisher` and why `n_repairs` was changed to `repairs`.

Reply:
The first observation is that if you have repairs (say) `[Insert, Delete]` then their score will be the same no matter how many `Shift`s there are. So `score([Insert, Delete]) == score([Insert, Delete, Shift, Shift])`.

The second observation is that `n_repairs` only differs from `repairs` in the number of `Shift`s. So, due to observation one, `score(repairs) == score(n_repairs)` is always true. Hence we can hoist the two `score` calls outside of the branch and compress them into one.

Reply:
That makes sense.